src/workerd/api/memory-cache.c++

```diff
@@ -239,14 +242,22 @@ SharedMemoryCache::Use::~Use() noexcept(false) {
 }
 
 kj::Maybe<kj::Own<CacheValue>> SharedMemoryCache::Use::getWithoutFallback(
-    const kj::String& key) const {
-  auto data = cache->data.lockExclusive();
+    const kj::String& key, SpanBuilder& span) const {
+  kj::Locked<ThreadUnsafeData> data = [&] {
+    auto memoryCacheLockRecord =
+        ScopedDurationTagger(span, "memory_cache_lock_wait_time_ns"_kjc, cache->timer);
```

Comment:
since this is used twice lets extract single `constexpr memoryCachekLockWaitTimeTag = ...`